pkg/docker/cockpit-atomic-storage

```diff
@@ -0,0 +1,501 @@
+                                   "sort_index": int(d.get('MAJOR', "0"))*100000 + int(d.get('MINOR', "0")),
+                                   "name": drive_name(d),
+                                   "class": drive_class(d),
+                                   "size": int(d.attr("size"))*512 }
+
+        (driver, lvol, vgroup) = get_driver_info()
+
+        pool_devices = [ ]
+        pool_drives = { }
+        extra_devices = [ ]
+
+        if vgroup:
+            other_parents = set()
+            for l in list_lvs(vgroup):
+                if l != lvol:
+                    other_parents |= set(list_parents(query_lvs(l, vgroup, "lv_dmpath")[0]))
+
+            for pv in list_pvs(vgroup):
+                for p in list_parents(pv):
+                    if p in drives:
+                        pool_drives[p] = True
+                        pool_devices.append({ "path": pv,
+                                              "sort_index": drives[p]["sort_index"],
+                                              "name": drives[p]["name"],
+                                              "class": drives[p]["class"],
+                                              "size": drives[p]["size"],
+                                              "shared": pv in other_parents })
+
+        for d in drives:
+            if d not in pool_drives:
+                extra_devices.append(drives[d])
+
+        if driver == "thinpool":
+            total = get_adjusted_thinpool_size(lvol, vgroup)
+            dm_path = query_lvs(lvol, vgroup, "lv_dm_path")[0]
+            used = get_devicemapper_thinpool_usage(dm_path)
+        else:
+            used = get_overlayfs_usage() + get_loopback_usage() + get_graph_usage()
+            fs_stat = os.statvfs("/var")
+            fs_used = (fs_stat.f_blocks - fs_stat.f_bfree)*fs_stat.f_frsize
+            if lvol:
+                fs_total = get_adjusted_thinpool_size(lvol, vgroup)
+            else:
+                fs_total = fs_stat.f_blocks * fs_stat.f_frsize
+            total =  fs_total - fs_used + used
+
+        return { "loopback": os.path.exists("/var/lib/docker/devicemapper/devicemapper/data"),
+                 "can_manage": can_manage and vgroup,
+                 "total": total,
+                 "used": used,
+                 "pool_devices": pool_devices,
+                 "extra_devices": extra_devices }
+
+    except:
+        return None
+
+def print_info(data):
+    if data:
+        sys.stdout.write(json.dumps(data) + "\n")
+        sys.stdout.flush()
+
+def cmd_monitor():
+    global can_manage
+    try:
+        with open("/dev/null", "w") as dev_null:
+            can_manage = call([ "atomic", "storage", "--help" ], stdout=dev_null) == 0
+    except:
+        can_manage = False
+
+    old_info = get_info()
+    print_info(old_info)
+
+    mon = subprocess.Popen([ "stdbuf", "-o", "L", "udevadm", "monitor", "-u", "-s", "block"],
+                           bufsize=1, stdout=subprocess.PIPE)
```

Comment:
Why do we need both bufsize=1 and stdbuf here? Doesn't bufsize=1 turn on line buffering already?